test-startup/src/test/java/StartupTest.java

```diff
@@ -31,50 +32,58 @@
-        try (Neo4jContainerExtension neo4jContainer = createEnterpriseDB(APOC_FULL, !TestUtil.isRunningInCI())
-                .withNeo4jConfig("dbms.transaction.timeout", "5s")) {
-
-            neo4jContainer.start();
-            assertTrue("Neo4j Instance should be up-and-running", neo4jContainer.isRunning());
-
-            Session session = neo4jContainer.getSession();
-            int procedureCount = session.run("SHOW PROCEDURES YIELD name WHERE name STARTS WITH 'apoc' RETURN count(*) AS count").peek().get("count").asInt();
-            int functionCount = session.run("SHOW FUNCTIONS YIELD name WHERE name STARTS WITH 'apoc' RETURN count(*) AS count").peek().get("count").asInt();
-            int coreCount = session.run("CALL apoc.help('') YIELD core WHERE core = true RETURN count(*) AS count").peek().get("count").asInt();
-
-            assertTrue(procedureCount > 0);
-            assertTrue(functionCount > 0);
-            assertTrue(coreCount > 0);
-        } catch (Exception ex) {
-            // if Testcontainers wasn't able to retrieve the docker image we ignore the test
-            if (TestContainerUtil.isDockerImageAvailable(ex)) {
-                ex.printStackTrace();
-                fail("Should not have thrown exception when trying to start Neo4j: " + ex);
+        for (var version: Neo4jVersion.values()) {
+            try (Neo4jContainerExtension neo4jContainer = createDB(version, APOC_FULL, !TestUtil.isRunningInCI())
+                    .withNeo4jConfig("dbms.transaction.timeout", "5s")) {
+
+                neo4jContainer.start();
+                assertTrue("Neo4j Instance should be up-and-running", neo4jContainer.isRunning());
+
+                Session session = neo4jContainer.getSession();
+                int procedureCount = session.run("SHOW PROCEDURES YIELD name WHERE name STARTS WITH 'apoc' RETURN count(*) AS count").peek().get("count").asInt();
+                int functionCount = session.run("SHOW FUNCTIONS YIELD name WHERE name STARTS WITH 'apoc' RETURN count(*) AS count").peek().get("count").asInt();
+                int coreCount = session.run("CALL apoc.help('') YIELD core WHERE core = true RETURN count(*) AS count").peek().get("count").asInt();
+
+                assertTrue(procedureCount > 0);
+                assertTrue(functionCount > 0);
+                assertTrue(coreCount > 0);
+            } catch (Exception ex) {
+                // if Testcontainers wasn't able to retrieve the docker image we ignore the test
+                if (TestContainerUtil.isDockerImageAvailable(ex)) {
+                    ex.printStackTrace();
+                    fail("Should not have thrown exception when trying to start Neo4j: " + ex);
+                } else if (!TestUtil.isRunningInCI()) {
+                    fail( "The docker image " + TestContainerUtil.neo4jEnterpriseDockerImageVersion + " should be available in the CI");
+                }
             }
         }
     }
```

Comment:
Refactor this in other branches to also have tests for community version (to avoid binary incompatibilities ever again)